bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1623,13 +1624,10 @@ OperationResult PreProcessor::launchReqPreProcessing(const PreProcessRequestMsgS
   requestsHandler_.preExecute(request, std::nullopt, cid, span);
   auto preProcessResult = static_cast<OperationResult>(request.outExecutionStatus);
   resultLen = request.outActualReplySize;
-  if (preProcessResult != OperationResult::SUCCESS) {
-    LOG_ERROR(logger(),
-              "Pre-execution failed" << KVLOG(
-                  cid, clientId, reqOffsetInBatch, reqSeqNum, (uint32_t)preProcessResult, resultLen));
-    return preProcessResult;
-  }
+
   if (request.outActualReplySize == 0) {
+    strcpy(preProcessResultBuffer, "Executed data is empty");
+    resultLen = strlen(preProcessResultBuffer);
     preProcessResult = OperationResult::EXEC_DATA_EMPTY;
     LOG_ERROR(logger(),
               "Pre-execution failed" << KVLOG(cid, clientId, reqOffsetInBatch, reqSeqNum, (uint32_t)preProcessResult));
```

Comment:
Why are you returning from here? In my last comment, I asked you to add the block to the end of the buffer in case of this error, too.